readthedocs/projects/models.py

```diff
@@ -621,7 +621,10 @@ def get_production_media_url(self, type_, version_slug, full_path=True):
         except NoReverseMatch:
             return ''
         if full_path:
-            path = '//{}{}'.format(settings.PRODUCTION_DOMAIN, path)
+            # Use project domain for full path --same domain as docs
+            # (project-slug.{PUBLIC_DOMAIN} or docs.project.com)
+            domain = self.subdomain()
+            path = f'//{domain}{path}'
```

Comment:
I think perhaps we should not use URL reversing here if it means requiring the main site to have an invalid URL in its URLConf. I think generating the URL directly is better than hacking the URLConf.

Reply:
I made this change to not use `reverse` and generate this URL manually.